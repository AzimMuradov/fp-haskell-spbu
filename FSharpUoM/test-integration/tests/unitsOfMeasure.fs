[<Measure>] type m;;
// Right: 
// [<Measure>]
// type m = m 


[<Measure>] type sec;
[<Measure>] type smt = sec^-3;

let m = (5<sec> * 12<sec> + 44<sec^2>) / 1<sec^5>;
m + 12<smt>;; // Right: "116</(sec^-3)>"

[<Measure>] type kg;
let f (x:int<kg>) = x + 15<kg>;; // Right: "int<kg> -> int<kg>"

// Physical formulas

[<Measure>] type vatt;
[<Measure>] type sec;
[<Measure>] type joule;


let Awork = 10<joule>;
let Ttime = 5<sec>;
let Npower = Awork / Ttime;; // Right: "2<joule/(sec)> "

[<Measure>] type kg;
[<Measure>] type m;
[<Measure>] type H;

let g = 9.8<H/kg>;
let mass = 2542.2423<kg>;
let h = 25687.123454<m>;
let PEnergy = g * mass * h;
let smt = PEnergy / 2.5;; // Right: "2.5598733589551717e8<H m>"