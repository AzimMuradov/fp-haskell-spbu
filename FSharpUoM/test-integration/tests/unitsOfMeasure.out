[<Measure>]
type m = m
val it: int</(sec^-3)> = 116
val f: int<kg> -> int<kg>
val Npower: int<joule/(sec)> = 2
val smt: double<H m> = 2.5598733589551717e8
